modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/h2/TableWithImplicitPkTest.java

```diff
@@ -0,0 +1,300 @@
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+import javax.cache.Cache;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.cache.affinity.rendezvous.RendezvousAffinityFunction;
+import org.apache.ignite.cache.query.FieldsQueryCursor;
+import org.apache.ignite.cache.query.SqlFieldsQuery;
+import org.apache.ignite.cache.query.annotations.QuerySqlField;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.DataRegionConfiguration;
+import org.apache.ignite.configuration.DataStorageConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.processors.query.IgniteSQLException;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.G;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.internal.processors.query.h2.H2Utils.IGNITE_SQL_ALLOW_IMPLICIT_PK;
+import static org.apache.ignite.testframework.GridTestUtils.assertThrows;
+
+/**
+ * Basic tests to check the possibility of creating tables without specifying a primary key.
+ */
+public class TableWithImplicitPkTest extends GridCommonAbstractTest {
+    /** Nodes count. */
+    private static final int NODES_CNT = 3;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super.getConfiguration(igniteInstanceName)
+            .setCacheConfiguration(cacheCfg())
+            .setDataStorageConfiguration(
+                new DataStorageConfiguration()
+                    .setDefaultDataRegionConfiguration(
+                        new DataRegionConfiguration()
+                            .setPersistenceEnabled(true)
+                    )
+            );
+    }
+
+    /**
+     * @return Cache configuration.
+     */
+    private CacheConfiguration<?, ?> cacheCfg() {
+        return new CacheConfiguration<>(DEFAULT_CACHE_NAME)
+            .setAffinity(new RendezvousAffinityFunction(false, 16))
+            .setBackups(1);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        cleanPersistenceDir();
+
+        startGridsMultiThreaded(NODES_CNT)
+            .addCacheConfiguration(cacheCfg());
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        List<String> cachesToDestroy = grid(0).cacheNames().stream()
+            .filter(name -> !DEFAULT_CACHE_NAME.equals(name))
+            .collect(Collectors.toList());
+
+        if (!cachesToDestroy.isEmpty()) {
+            grid(0).destroyCaches(cachesToDestroy);
+            awaitPartitionMapExchange();
+        }
+
+        System.setProperty(IGNITE_SQL_ALLOW_IMPLICIT_PK, "true");
+    }
+
+    @Test
+    public void testBasicOperations() {
```

Comment:
let's add two more scenarios:
1) `SELECT *` returns all columns but implicit pk. `SELECT _key, *` returns all columns including implicit pk
2) it's possible to get value from cache by pk returned by `SELECT _key, *` query. Moreover, values returned by select match the values returned by cache for the same key